dev/cognitiveservices/data-plane/Language/common.json

```diff
@@ -722,6 +722,108 @@
           "minimum": 1
         }
       }
+    },
+    "SummarizationTaskParametersBase": {
+      "type": "object",
+      "description": "Supported parameters for both Abstractive Summarization and Extractive Summarization task.",
+      "properties": {
+        "sentenceCount": {
+          "type": "integer",
+          "default": 3
+        },
+        "stringIndexType": {
+          "$ref": "#/definitions/StringIndexType"
+        }
+      }
+    },
+    "AbstractiveSummarizationTaskParametersBase": {
+      "type": "object",
+      "description": "Supported parameters for an Abstractive Summarization task.",
+      "properties": {
+        "phraseControls": {
+          "type": "array",
+          "description": "Control the phrases to be used in the summary.",
+          "items": {
+            "$ref": "#/definitions/PhraseControl"
+          }
+        },
+        "labFeatures": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Used for experimental features."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/SummarizationTaskParametersBase"
+        }
+      ]
+    },
+    "PhraseControl": {
+      "type": "object",
+      "description": "Control the phrases to be used in the summary.",
+      "required": [
+        "targetPhrase",
+        "strategy"
+      ],
+      "properties": {
+        "targetPhrase": {
+          "type": "string",
+          "description": "The target phrase to control."
+        },
+        "strategy": {
+          "$ref": "#/definitions/PhraseControlStrategy"
+        }
+      }
+    },
+    "PhraseControlStrategy": {
+      "enum": [
+        "encourage",
+        "discourage",
+        "disallow"
+      ],
+      "type": "string",
+      "description": "The strategy to use in phrase control.",
+      "x-ms-enum": {
+        "modelAsString": false,
+        "name": "PhraseControlStrategy",
+        "values": [
+          {
+            "value": "encourage",
+            "description": "The model is more favor to select the target phrase in the summary if there are multiple alternate."
+          },
+          {
+            "value": "discourage",
+            "description": "The model is less favor to select the target phrase in the summary if there are multiple alternate."
+          },
+          {
+            "value": "disallow",
+            "description": "The model will avoid to select the target phrase in the summary if that phrase is not in the input."
+          }
+        ]
+      }
+    },
+    "SummaryContext": {
+      "type": "object",
+      "description": "The context of the summary.",
+      "required": [
+        "offset",
+        "length"
+      ],
+      "properties": {
```

Comment:
Should we include encoding?

Reply:
The encoding is included in common.json#/definitions/StringIndexType, which is specified from the request. See SummarizationTaskParametersBase.